@market/hooks/usePrivateAskTransaction.ts

```diff
@@ -0,0 +1,117 @@
+import { useContractTransaction } from '@shared'
+import { useState } from 'react'
+import { useContractContext } from '@market/providers'
+import { NFTObject } from '@zoralabs/nft-hooks'
+import { ContractTransaction } from 'ethers'
+import { parseUnits } from '@ethersproject/units'
+import { usePrivateAskContext } from '@market/providers/PrivateAskProvider'
+
+export const CREATE_ASK: string = 'createPrivateAsk'
+export const CANCEL_ASK: string = 'cancelPrivateAsk'
+export const FILL_ASK: string = 'fillPrivateAsk'
+
+type PrivateAskTransaction = typeof CREATE_ASK | typeof CANCEL_ASK | typeof FILL_ASK
+
+interface CreateAskTxValues {
+  price?: string
+  buyerAddress?: string
+}
+
+interface usePrivateAskTransactionProps {
+  nft: NFTObject
+  collectionAddress?: string
+  tokenId?: string
+  onNext?: () => void
+}
+
+export const usePrivateAskTransaction = ({
+  nft: nftData,
+  onNext,
+}: usePrivateAskTransactionProps) => {
+  const { PrivateAsks } = useContractContext() // Should this all be moved to usePrivateAskContext?
```

Comment:
worth having a conversation about this stuff, different approaches have different trade-offs